cloudinit/config/schema.py

```diff
@@ -715,6 +716,23 @@ def _sort_property_order(value):
     return 0
 
 
+def _flatten(xs):
+    for x in xs:
+        if isinstance(x, Iterable) and not isinstance(x, (str, bytes)):
+            yield from _flatten(x)
+        else:
+            yield x
+
+
+def _collect_subschema_types(property_dict: dict, multi_key: str) -> List[str]:
+    property_types = [
+        subschema["type"]
+        for subschema in property_dict.get(multi_key, {})
+        if subschema.get("type") and not subschema.get("deprecated", False)
+    ]
```

Comment:
I think we need to also take into account 'enum' keys too. Maybe something like the following:

```suggestion
    property_types = []                                                        
    for subschema in property_dict.get(multi_key, {}):                         
        if subschema.get(DEPRECATED_KEY):  # don't document deprecated types   
            continue                                                           
        if subschema.get("enum"):                                              
            property_types.extend(                                             
                [                                                              
                    f"``{_YAML_MAP.get(enum_value, enum_value)}``"             
                    for enum_value in subschema.get("enum", [])                
                ]                                                              
            )                                                                  
        elif subschema.get("type"):                                            
            property_types.append(subschema["type"])      
```

Reply:
Done.